algosdk/v2client/algod.py

```diff
@@ -126,6 +126,35 @@ def application_info(self, application_id, **kwargs):
         req = "/applications/" + str(application_id)
         return self.algod_request("GET", req, **kwargs)
 
+    def application_box_by_name(
+        self, application_id: int, box_name: bytes, **kwargs
+    ):
+        """
+        Return the value of an application's box.
```

Comment:
This endpoint (and the other 3) returns a dictionary containing base64 encoded values. This behavior is likely going to confuse consumers.

At the bare minimum, we should better document what the response looks like and perhaps even provide an example.

Though I'm debating whether we should actually process the response and decode the values into bytes, similar to what the JS SDK now does.

Reply:
After discussion with @winder and @michaeldiamant, we've decided to leave the behavior of these endpoints as-is for this SDK, while at the same time retaining the new behavior in the JS SDK.

Some of the factors that went into this decision were:
* The Python SDK lacks any code generation for response models. Returning a typed response object first involves creating a definition for that object, which is a considerable undertaking, especially if we need to maintain these definitions by hand.
* Type annotations are sparse in this SDK, and in fact it does not run `mypy` or a similar static analyzer. This diminishes the potential advantages of having typed response objects.
* Currently the Python SDK has a high level of consistency and simplicity, especially with response handling. A deviation for these endpoints would reduce those properties, and therefor reduce the maintainability of the code.

Generally speaking we are interesting in improving the ergonomics of this SDK, but at this time the effort needed is too high.

Reply:
I opened #388 to resolve this by mentioning box names/values will be base64 encoded in responses